enhancements/monitoring/metrics-collection-profiles.md

```diff
@@ -0,0 +1,430 @@
+      k8s-app: telemeter-client
+  metricRelabelings:
+  - sourceLabels: [__name__]
+    action: keep
+    regex: "federate_samples|federate_filtered_samples"
+ ```
+
+Note: 
+- the `metricRelabeling` section only keeping two metrics, while the rest is
+dropped.
+- the metrics in the `keep` section were obtained with the help of a script that
+  parsed all Alerts, PrometheusRules and Console dashboards to determine what
+  metrics were actually being used.
+
+Finally, a team that adopts the metrics collection profile feature should also 
+add themselves and their component to the list under 
+[Infrastruture needed](#infrastructure-needed-optional)
+
+#### Variation [optional]
+
+NA
+
+### API Extensions
+
+- Addition of a new field to the `cluster-monitoring-config` ConfigMap
+
+```go
+type PrometheusK8sConfig struct {
+	// Defines the metrics collection profile that Prometheus uses to collect
+	// metrics from the platform components. Supported values are `full` or
+	// `minimal`. In the `full` profile (default), Prometheus collects all
+	// metrics that are exposed by the platform components. In the `minimal`
+	// profile, Prometheus only collects metrics necessary for the default
+	// platform alerts, recording rules, telemetry and console dashboards.
+	CollectionProfile CollectionProfile `json:"collectionProfile,omitempty"`
+}
+```
+
+### Implementation Details/Notes/Constraints
+
+Each OpenShift team that wants to adopt this feature will be responsible for
+providing the different monitors. This work is not trivial. Dependencies between
+operators and their metrics exist. This makes it difficult for developers to
+determine whether a given metric can be excluded from the `minimal` profile or
+not. To aid teams with this effort the monitoring team will provide:
+- a CLI tool that consumes a ServiceMonitor and a list of metrics and
+  generates the `minimal` ServiceMonitor for that operator.
+- an origin/CI test that validates for all Alerts and PrometheusRules that the
+  metrics used by them are present in the `keep` expression of the
+  ServiceMonitor for the `minimal` profile
+
+
+### Risks and Mitigations
+
+- How are monitors supposed to be kept up to date? In 4.12 a metric that wasn't
+  being used in an alert is now required, how does the monitor responsible for
+  that metric gets updated?
+  - The origin/CI test mentioned in the previous section will fail if there is a
+    resource (Alerts/PrometheusRules/Dashboards) using a metric which is not
+    present in the ServiceMonitor;
+
+- A new profile is added, what will happen to operators that had implemented
+  metrics collection profiles but did not implement the latest profile.
+  - The monitoring team will use the list under 
+  [Infrastruture needed](#infrastructure-needed-optional) to help the developers
+  with the addoption of the new metrics collection profile.
+  - Furthermore, we plan on adding extra origin/CI tests to validate each new
+    profile if applicable.
```

Comment:
The general idea around this currently (in addition to above), is that in the absence of a profile implementation by an operator (if the cluster admin switches to that profile) none of the metrics will be exported out (the assumption is that they want to drop everything in case of `minimal`).

Reply:
Removed this part of this section. Furthermore since we now only support two profiles and it's out of the goal of adding further profiles I feel like this is not relevant anymore.